### DISCUSSION TITLE
Using PHP attributes feature

### DISCUSSION THREAD

**Discussion by SaeedDev94** [laravel/framework#40319] [Ideas] [answered]
Hi,
It would be great if we can use PHP attributes in controller.
For example:
Imagine if we can add middleware to method using attribute:
```php
#[Middleware(UserHasRole::class, ['blog_editor'])]
function method () {}
```
What do you think ?

**Comment by SaeedDev94** [comment, marked as answer]
I just published [`Laravel #[Annotation]`](https://github.com/Saeed-Pooyanfar/laravel-annotation) package for this idea.

**Comment by zyz954489346** [comment]
I think is a good idea , especially route register.
